ssl/ssl_test.cc

```diff
@@ -11999,5 +11999,24 @@ TEST(SSLTest, SSLFileTests) {
   ASSERT_EQ(remove(ecdsa_pem_filename), 0);
 }
 
+TEST(SSLTest, IncompatibleTLSVersionState) {
+  // Using the following ASN.1 DER Sequence where 42 is the serialization
+  // format version number of some future version not currently supported:
+  // SEQUENCE {
+  //   SEQUENCE {
+  //     INTEGER { 42 }
+  //   }
+  // }
+  static constexpr size_t INCOMPATIBLE_DER_LEN = 7;
+  static const uint8_t INCOMPATIBLE_DER[INCOMPATIBLE_DER_LEN] = {
+      0x30, 0x05, 0x30, 0x03, 0x02, 0x01, 0x2a};
+
+  bssl::UniquePtr<SSL_CTX> ctx(SSL_CTX_new(TLS_method()));
+  ASSERT_TRUE(ctx);
+
+  ASSERT_FALSE(
+      SSL_from_bytes(INCOMPATIBLE_DER, INCOMPATIBLE_DER_LEN, ctx.get()));
+}
```

Comment:
For V1177861396 is it possible to verify this test is failing due to the version mismatch and not something like INCOMPATIBLE_DER doesn't have enough data to read the state from?

Reply:
I'll have to assign explicit error codes for this to do so, I've verified in a debugger that this is the reason it is failing. But let me explore just mapping new error types in.